internal/controllers/sync-volume/utils.go

```diff
@@ -59,9 +61,15 @@ func getConverter(vol *stosv1.Volume, obj storageos.Object) func() error {
 		}
 
 		var replicas *uint8
-		if stosVol.Replicas != nil {
-			r := uint8(len(*stosVol.Replicas))
-			replicas = &r
+		replicaCount, ok := stosVol.Labels[storageos.ReservedLabelReplicas]
+		if ok {
+			count, err := strconv.Atoi(replicaCount)
+			if err == nil {
```

Comment:
Could you please at least log if conversion fails? For me this is a bit optimistic solution :)